### PR TITLE
Add PrivateSpaceOwner to SpaceResource

### PR DESCRIPTION
**Pull request by iryan2**
Related to work in this [PR](https://github.com/OctopusDeploy/OctopusDeploy/pull/11641)

### CODE REVIEW

**Comment by iryan2** [comment]
@andrewabest Is there anything else we should be including in this change? I hadn't planned on doing this immediately, but I believe I need it for the change to the [test we spoke about](https://github.com/OctopusDeploy/OctopusDeploy/pull/11448/files#r827614256), or at least doing it up front will prevent a temporary workaround.

**Comment by andrewabest** [comment]
> @andrewabest Is there anything else we should be including in this change? I hadn't planned on doing this immediately, but I believe I need it for the change to the [test we spoke about](https://github.com/OctopusDeploy/OctopusDeploy/pull/11448/files#r827614256), or at least doing it up front will prevent a temporary workaround.

No this seems fine. It isn't a writeable property, which we want, and it will either be null if no value exists, or have the appropriate value within it. Looks good.

File: source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETCore.approved.txt [hunk 1]
```diff
@@ -4900,6 +4900,7 @@ Octopus.Client.Model
     Boolean IsDefault { get; set; }
     Boolean IsPrivate { get; set; }
```
Comment (andrewabest) [inline source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETCore.approved.txt:4901]: I suspect there is also the netframework approval, not sure how you generated that last time though.

**Review by andrewabest — APPROVED** [review]
Looks good

File: source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETCore.approved.txt (hunk 1 above)
Comment (iryan2) [reply source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETCore.approved.txt:4901]: Yeah, [last time](https://github.com/OctopusDeploy/OctopusClients/pull/639) I updated two approved files, this time I was only prompted for one. The test passes whether I include it or not, and it's passed all the tests here. Should I be concerned or just go ahead and merge it?

Comment (andrewabest) [reply source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETCore.approved.txt:4901]: Last time you wouldn't have been prompted for the netframework file - you are on a mac, and so can't run that test. You must have updated it by hand. I would do the same here, update the netframework file by hand to ensure it matches what we would expect to be the output.

Comment (iryan2) [reply source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETCore.approved.txt:4901]: Ah sure, I guess it was a CI failure that made me update it last time. Which begs the question, why did it pass without it? Anyway, I've updated it it's passed again so I suppose I'll merge this.

**Pull request merged**